src/Microsoft.PowerShell.Commands.Management/commands/management/ComputerUnix.cs

```diff
@@ -153,13 +153,13 @@ protected override void StopProcessing()
         /// <summary>
         /// Run a command.
         /// </summary>
-        protected void RunCommand(String command, String args) {
+        protected void RunCommand(String command = "/sbin/shutdown", String args = "") {
```

Comment:
I suggest don't complicate the PR and only fix the code with right `shutdown` for Linux and Mac. Then open new issue to enhance the cmdlet to support `Force` (and grace shutdown) on Unix.

Reply:
Good point, @iSazonov - I've opened:
* #19791

Reply:
@iSazonov defaults have been removed.  I had put them in there just because those were the previous hardcoded values.  Changed `String `to `string `per your request.

Reply:
Thanks, @mark-wilson-spc; note that changing 

```c#
            if (Platform.IsMacOS)
            {
                args = "now";
            }
```

to

```c#
            if (Platform.IsMacOS)
            {
                args = "-h now";
            }
```

is still necessary.

Reply:
Mac `Stop-Computer` args have been changed from `now` to `-h now` .

Reply:
@mklement0 changing the args caused the unelevated pester tests to fail.

Reply:
@mark-wilson-spc,  it is a fundamental requirement on macOS to run `/sbin/shutdown` via `sudo`. In other words: the test is flawed.

Reply:
@mark-wilson-spc, glancing at the tests, I would have expected them to fail previously as well, given that unelevated invocation with _any_ arguments - including none, as currently - causes `/sbin/shutdown` to fail categorically on macOS, printing `shutdown: NOT super-user` to stderr and terminating with exit code `1`.

Reply:
@iSazonov, not via `/sbin/shutdown`, specifically, but they _can_ via the GUI and via the AppleScript commands proposed for _graceful_ shutdown / restart proposed in #19791. 

That is, in the latter two cases _no_ administrator role / `sudo` calls are required.

Reply:
So do we need to run the test with elevated rights?

Reply:
Are we open to narrowing this to an atomic change implementing the restart fix for Linux distros, and then moving the outstanding MacOS issues to @mklement0 's [#19791](https://github.com/PowerShell/PowerShell/issues/19791)?

Reply:
@iSazonov, I haven't dug too deep, but it looks like the `RunCommand()` call is by design never reached  when the test hooks are in place, which indeed makes it mysterious that changing the arguments should make any difference.

It looks like the existing tests are virtually useless on Unix.
To implement meaningful tests (that do not actually _invoke_ `/sbin/shutdown`) would require testing for being `root` and failing if not - both on macOS and Linux. (If the graceful methods are implemented for macOS, that would have to be taken into account.)

Given the lack of actual invocation, the syntax of the arguments can only be verified by comparing the dynamically constructed command line to what it _should be_, as determined via the `man` pages and as verified by interactive invocation (it's safe to assume that this syntax will never change).

Reply:
@mark-wilson-spc, note that the _restart_ fix would work for macOS too, but _shutdown_ would still be broken, if `"now"` isn't changed to `"-h now"`.

My personal preference is to include the latter change, and then deal with the following aspects separately:
* graceful shutdown/reboot on macOS: #19791
* inability to programmatically detect failure: #19802

Given my discussion of the tests above: Is it possible that the macOS test failure you saw was _transient_, unrelated to your changes? If so, including the `"-h now"` too shouldn't be a problem.

Reply:
Thank you for the clarification, @mklement0 .  My thought was that we could go ahead and push out the working fix (`Restart-Computer` for Linux and MacOs) immediately since it is ready to go and has a broad benefit, and then address the thornier shutdown issue separately.

As far as the testing goes, I am new at this and am not in a position to speak to the results of the automated testing (especially where MacOS is concerned).  I had assumed all tests were valid and robust, and that it was necessary to pass all tests for merge approval.  I'm not sure who can best speak to your question regarding the possible transient or false nature of a test result.

Reply:
Thanks, @mark-wilson-spc - I'll have to defer to others too with respect to the tests. 

As for whether to hold off on the macOS `"-h now"` fix to also make `Stop-Computer` work:

* As noted, even without the fix, `Restart-Computer` _will_ start to work on macOS, because `-r now` works on macOS too.

* If we proceed without the shutdown fix for macOS, we'll have a _somewhat_ working `Restart-Computer` there - one that forces  _instant shutdown with potential data loss_ -  and a still broken `Stop-Computer`. 

* If we include the shutdown fix for macOS we'll have _somewhat_ working `Restart-Computer` and `Stop-Computer` cmdlets on macOS, with the caveats noted.

If _either_ approach gets into a stable release _before_ the graceful approach is implemented on macOS (I'm not holding my breath), we'll have a backward-compatibility problem on our hand: users may by then have come to _rely_ on the forceful and therefore _guaranteed_ shutdown/reboot, even without having specified `-Force`.

Therefore, the most conservative approach would be: fix the problem _for Linux only_ for now (which already had a working `Stop-Computer`, but a broken `Restart-Computer`), which would require deliberately setting the arguments to pass to `shutdown` to something _invalid_ for `Restart-Computer` too.

I hope that lays out the options clearly, and I'll leave the decision to others.

Reply:
In any case we can not really stop/reboot CI system and have to use test mosc/hook. **So we must check the cmdlets manually before merge.**

As for code, I suggest to implement "-h now" for macOs as @mklement0 proposed. As result after the PR we should get the cmdlets working well on Linux and macOS but without (1) grace scenario, (2) error processing - both are for follow PRs.

Reply:
I am working on an implementation and should have it ready later today.